modules/fedimint-walletv2-common/src/esplora_api.rs

```diff
@@ -0,0 +1,53 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use bitcoin::{Address, Txid};
+use fedimint_core::util::SafeUrl;
+use fedimint_core::{apply, async_trait_maybe_send, dyn_newtype_define};
+use serde::{Deserialize, Serialize};
+
+#[apply(async_trait_maybe_send!)]
+pub trait IEsploraConnection: Debug {
+    async fn get_address_utxo(
+        &self,
+        esplora: SafeUrl,
+        address: Address,
+    ) -> anyhow::Result<Vec<AddressUnspentTxOut>>;
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AddressUnspentTxOut {
+    pub txid: Txid,
+    pub vout: u32,
+    pub value: u64,
+    pub status: AddressUnspentTxOutStatus,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AddressUnspentTxOutStatus {
+    pub block_height: Option<u64>,
+}
+
+dyn_newtype_define! {
+    #[derive(Clone)]
+    pub DynEsploraConnection(Arc<IEsploraConnection>)
+}
+
+#[derive(Debug, Clone)]
+pub struct RealEsploraConnection;
```

Comment:
We might want to stuff `reqwest::Client` here to reuse it.

> The Client holds a connection pool internally, so it is advised that you create one and reuse it.

https://docs.rs/reqwest/latest/reqwest/struct.Client.html